File: core/calcST.jl
module CalcST

include("hydro_model_structs.jl")
using .HydroModelStructs

export calcST


# This module calculates storativity (S) and transmissivity (T) from the given hydro parameters
# Inputs: h: reservoir thickness in feet
#         phi: porosity fraction (given in percentage ex. 10 = 10%)
#         kap: intrinsic permeability in millidarcy
#         params::HydroModelStructs.HydroParameters: Struct containing other hydro parameters.

# Outputs: S: storativity
#          T: transmissivity
function calcST(h::Float64, phi::Float64, kap::Float64, params::HydroModelStructs.HydroParameters)
    # convert permeability (mD -> m^2)
    kap_m2 = kap * 10^(-3) * 9.9e-13 # manually entered data (-perm)

    rho = params.rho# fluid density
    g = params.g # accl gravity
    beta = params.beta # fluid compressibility
    alphav = params.alphav # vertical compressibility of aquifer
    mu = params.mu # dynamic viscosity

    # calculate storativity
    h_meters = h * 0.3048 # convert feet to meters
    S = rho * g * h_meters * (alphav + (phi * 1e-2) * beta)
    
    # calculate transmissivity
    K = kap_m2 * rho * g / mu # Saturated hydraulic conductivity
    T = K * h_meters

    return S, T
    
end



# HARDCODED DATA FOR TESTING ---------------------------------------------------------------------------------------


function test_calcST()
    # mock HydroParameters struct
    params = HydroParameters(
        100.0,  # aquifer_thickness
        10.0,   # porosity_percent
        150.0,  # permeability
        1000.0, # rho
        9.81,   # g
        0.001,  # mu
        4.5e-10,# beta
        1.0e-9  # alphav
    )

    
    h = 100.0  
    phi = 10.0 
    kap = 150.0 

    S, T = calcST(h, phi, kap, params)
    println("Storativity (S): $S")
    println("Transmissivity (T): $T")
end

end # module


using .CalcST
CalcST.test_calcST()

File: FSP2.0beta_MatlabFiles/technical code/calcST.m
%Darren Pais
%Surveillance and Optimization, Reservoir
%2016+

% Outputs: S: Storativity
%          T: Transmissivity

% Inputs: h: reservoir thickness
%         phi: porosity fraction
%         kap: intrinsic permeability
%         valsarr: array of other inputs as below (to conform with rest of
%         code, a subset of these are used in the calc as below) 


function [S , T] = calcST(valsarr , h , phi , kap)

%all SI below with conversions as needed
rho = valsarr(5); %density
g = valsarr(6); %accl gravity
mu = valsarr(7); %dynamic viscosity
beta = valsarr(8); %fluid compressibility
alphav = valsarr(9); %vertical compressibility of aquifer

kap = kap*10^-3*9.9e-13; %convert permeability mD to m^2

S = rho*g*h*(alphav+phi*beta); %storativity 

% specific storage: (For MODFLOW Comparison
%  SpecificStorage = S./h  % rho*g*(alphav+phi*beta); % 

K = kap*rho*g/mu ;  %saturated hydraulic conductivity
T = K*h ;%transmissivity

end